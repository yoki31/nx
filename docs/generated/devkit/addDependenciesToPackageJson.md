# Function: addDependenciesToPackageJson

â–¸ **addDependenciesToPackageJson**(`tree`, `dependencies`, `devDependencies`, `packageJsonPath?`, `keepExistingVersions?`): [`GeneratorCallback`](../../devkit/documents/GeneratorCallback)

Add Dependencies and Dev Dependencies to package.json

For example:

```typescript
addDependenciesToPackageJson(tree, { react: 'latest' }, { jest: 'latest' });
```

This will **add** `react` and `jest` to the dependencies and devDependencies sections of package.json respectively.

#### Parameters

| Name                    | Type                                  | Description                                                                 |
| :---------------------- | :------------------------------------ | :-------------------------------------------------------------------------- |
| `tree`                  | [`Tree`](../../devkit/documents/Tree) | Tree representing file system to modify                                     |
| `dependencies`          | `Record`\<`string`, `string`\>        | Dependencies to be added to the dependencies section of package.json        |
| `devDependencies`       | `Record`\<`string`, `string`\>        | Dependencies to be added to the devDependencies section of package.json     |
| `packageJsonPath?`      | `string`                              | Path to package.json                                                        |
| `keepExistingVersions?` | `boolean`                             | If true, prevents existing dependencies from being bumped to newer versions |

#### Returns

[`GeneratorCallback`](../../devkit/documents/GeneratorCallback)

Callback to install dependencies only if necessary, no-op otherwise
