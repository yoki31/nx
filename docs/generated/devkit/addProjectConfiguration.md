# Function: addProjectConfiguration

â–¸ **addProjectConfiguration**(`tree`, `projectName`, `projectConfiguration`, `standalone?`): `void`

Adds project configuration to the Nx workspace.

#### Parameters

| Name                   | Type                                                                  | Default value | Description                                                             |
| :--------------------- | :-------------------------------------------------------------------- | :------------ | :---------------------------------------------------------------------- |
| `tree`                 | [`Tree`](../../devkit/documents/Tree)                                 | `undefined`   | the file system tree                                                    |
| `projectName`          | `string`                                                              | `undefined`   | unique name. Often directories are part of the name (e.g., mydir-mylib) |
| `projectConfiguration` | [`ProjectConfiguration`](../../devkit/documents/ProjectConfiguration) | `undefined`   | project configuration                                                   |
| `standalone`           | `boolean`                                                             | `true`        | whether the project is configured in workspace.json or not              |

#### Returns

`void`
