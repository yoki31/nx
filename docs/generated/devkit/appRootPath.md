# Variable: appRootPath

â€¢ `Const` **appRootPath**: `string` = `workspaceRoot`

The root of the workspace.

**`Deprecated`**

use workspaceRoot instead
