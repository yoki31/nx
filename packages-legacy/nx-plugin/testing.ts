export * from '@nx/plugin/testing';
