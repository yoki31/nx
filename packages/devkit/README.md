<p style="text-align: center;">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://raw.githubusercontent.com/nrwl/nx/master/images/nx-dark.svg">
    <img alt="Nx - Smart Monorepos · Fast CI" src="https://raw.githubusercontent.com/nrwl/nx/master/images/nx-light.svg" width="100%">
  </picture>
</p>

{{links}}

<hr>

# Nx: Smart Monorepos · Fast CI

Nx is a build system with built-in tooling and advanced CI capabilities. It helps you maintain and scale monorepos, both locally and on CI.

This package contains a set of utilities for creating Nx plugins.

{{content}}
