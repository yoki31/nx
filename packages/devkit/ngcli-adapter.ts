/**
 * @category Ng CLI Adapter
 */
// eslint-disable-next-line @typescript-eslint/no-restricted-imports
export {
  wrapAngularDevkitSchematic,
  mockSchematicsForTesting,
  NxScopedHost,
} from 'nx/src/adapter/ngcli-adapter';
