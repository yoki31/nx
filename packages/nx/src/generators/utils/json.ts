import type { Tree } from '../tree';
import { parseJson, serializeJson } from '../../utils/json';
import type { JsonParseOptions, JsonSerializeOptions } from '../../utils/json';

/**
 * Reads a json file, removes all comments and parses JSON.
 *
 * @param tree - file system tree
 * @param path - file path
 * @param options - Optional JSON Parse Options
 */
export function readJson<T extends object = any>(
  tree: Tree,
  path: string,
  options?: JsonParseOptions
): T {
  if (!tree.exists(path)) {
    throw new Error(`Cannot find ${path}`);
  }
  try {
    return parseJson(tree.read(path, 'utf-8'), options);
  } catch (e) {
    throw new Error(`Cannot parse ${path}: ${e.message}`);
  }
}

/**
 * Writes a JSON value to the file system tree

 * @param tree File system tree
 * @param path Path of JSON file in the Tree
 * @param value Serializable value to write
 * @param options Optional JSON Serialize Options
 */
export function writeJson<T extends object = object>(
  tree: Tree,
  path: string,
  value: T,
  options?: JsonSerializeOptions
): void {
  const serialized = serializeJson(value, options);
  tree.write(path, `${serialized}\n`);
}

/**
 * Updates a JSON value to the file system tree
 *
 * @param tree File system tree
 * @param path Path of JSON file in the Tree
 * @param updater Function that maps the current value of a JSON document to a new value to be written to the document
 * @param options Optional JSON Parse and Serialize Options
 */
export function updateJson<T extends object = any, U extends object = T>(
  tree: Tree,
  path: string,
  updater: (value: T) => U,
  options?: JsonParseOptions & JsonSerializeOptions
): void {
  const updatedValue = updater(readJson(tree, path, options));
  writeJson(tree, path, updatedValue, options);
}
