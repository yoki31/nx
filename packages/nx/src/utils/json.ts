import { parse, printParseErrorCode, stripComments } from 'jsonc-parser';
import type { ParseError, ParseOptions } from 'jsonc-parser';
import { LinesAndColumns } from 'lines-and-columns';
import { codeFrameColumns } from './code-frames';

export { stripComments as stripJsonComments };

export interface JsonParseOptions extends ParseOptions {
  /**
   * Expect JSON with javascript-style
   * @default false
   */
  expectComments?: boolean;
  /**
   * Disallow javascript-style
   * @default false
   */
  disallowComments?: boolean;

  /**
   * Allow trailing commas in the JSON content
   */
  allowTrailingComma?: boolean;
}

export interface JsonSerializeOptions {
  /**
   * the whitespaces to add as indentation to make the output more readable.
   * @default 2
   */
  spaces?: number;
}

/**
 * Parses the given JSON string and returns the object the JSON content represents.
 * By default javascript-style comments and trailing commas are allowed.
 *
 * @param input JSON content as string
 * @param options JSON parse options
 * @returns Object the JSON content represents
 */
export function parseJson<T extends object = any>(
  input: string,
  options?: JsonParseOptions
): T {
  try {
    if (options?.expectComments !== true) {
      return JSON.parse(input);
    }
  } catch {}

  options = { allowTrailingComma: true, ...options };

  const errors: ParseError[] = [];
  const result: T = parse(input, errors, options);

  if (errors.length > 0) {
    throw new Error(formatParseError(input, errors[0]));
  }

  return result;
}

/**
 * Nicely formats a JSON error with context
 *
 * @param input JSON content as string
 * @param parseError jsonc ParseError
 * @returns
 */
function formatParseError(input: string, parseError: ParseError) {
  const { error, offset, length } = parseError;
  let { line, column } = new LinesAndColumns(input).locationForIndex(offset);
  line++;
  column++;

  return (
    `${printParseErrorCode(error)} in JSON at ${line}:${column}\n` +
    codeFrameColumns(input, {
      start: { line, column },
      end: { line, column: column + length },
    }) +
    '\n'
  );
}

/**
 * Serializes the given data to a JSON string.
 * By default the JSON string is formatted with a 2 space indentation to be easy readable.
 *
 * @param input Object which should be serialized to JSON
 * @param options JSON serialize options
 * @returns the formatted JSON representation of the object
 */
export function serializeJson<T extends object = object>(
  input: T,
  options?: JsonSerializeOptions
): string {
  return JSON.stringify(input, null, options?.spaces ?? 2);
}
