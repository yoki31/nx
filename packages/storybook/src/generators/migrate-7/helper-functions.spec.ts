import {
  addProjectConfiguration,
  getPackageManagerCommand,
  output,
  ProjectConfiguration,
  readProjectConfiguration,
  Tree,
} from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import * as allProjects from './test-configs/all-projects.json';
import {
  addViteConfigFilePathInFrameworkOptions,
  changeCoreCommonImportToFramework,
  getAllStorybookInfo,
  logResult,
  onlyShowGuide,
  removePathResolvesFromNextConfig,
  removeTypecastFromMainTs,
  removeUiFrameworkFromProjectJson,
  removeViteTsConfigPathsPlugin,
} from './helper-functions';

describe('Helper functions for the Storybook 7 migration generator', () => {
  let tree: Tree;

  beforeEach(async () => {
    tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    addAllProjectsToWorkspace(tree);
  });

  describe('removePathResolvesFromNextConfig', () => {
    it(`should remove path from nextConfigPath in main.js`, async () => {
      const { content: content1 } = removePathResolvesFromNextConfig(
        tree,
        `apps/nextapp/.storybook/main.js`
      );
      const { content: content2 } = removePathResolvesFromNextConfig(
        tree,
        `apps/nextapp-ts/.storybook/main.ts`
      );

      expect(content1).toMatchSnapshot();

      expect(content2).toMatchSnapshot();
    });
  });

  describe('removeTypecastFromMainTs', () => {
    it(`should remove typecast from TypeScript files`, async () => {
      const { content: content1 } = removeTypecastFromMainTs(
        tree,
        `apps/rv2-ts/.storybook/main.ts`
      );
      const { content: content2 } = removeTypecastFromMainTs(
        tree,
        `apps/nextapp-ts/.storybook/main.ts`
      );

      expect(content1).toMatchSnapshot();

      expect(content2).toMatchSnapshot();
    });
  });

  describe('getAllStorybookInfo and onlyShowGuide', () => {
    let allStorybookInfo;
    beforeEach(() => {
      allStorybookInfo = getAllStorybookInfo(tree);
    });
    it('should return all info for all projects with Storybook', () => {
      expect(allStorybookInfo).toMatchSnapshot();
    });

    it('should onlyShowGuide and the correct instructions', () => {
      const outputSpy = jest.spyOn(output, 'log').mockImplementation();
      onlyShowGuide(allStorybookInfo);
      const pm = getPackageManagerCommand();
      expect(outputSpy).toHaveBeenCalledWith(
        expect.objectContaining({
          bodyLines: [
            'You can run the following commands manually to upgrade your Storybook projects to Storybook 7:',
            '',
            '1. Call the Storybook upgrade script:',
            `${pm.exec} storybook@latest upgrade`,
            '',
            '2. Call the Nx generator to prepare your files for migration:',
            'nx g @nx/storybook:migrate-7 --onlyPrepare',
            '',
            '3. Call the Storybook automigrate scripts:',
            'Run the following commands for each Storybook project:',
            `${pm.exec} storybook@latest automigrate --config-dir apps/nextapp/.storybook --renderer @storybook/react`,
            `${pm.exec} storybook@latest automigrate --config-dir apps/nextapp-ts/.storybook --renderer @storybook/react`,
            `${pm.exec} storybook@latest automigrate --config-dir apps/rv1/.storybook --renderer @storybook/react`,
            `${pm.exec} storybook@latest automigrate --config-dir apps/rv2-ts/.storybook --renderer @storybook/react`,
            `${pm.exec} storybook@latest automigrate --config-dir apps/rw1/.storybook --renderer @storybook/react`,
            `${pm.exec} storybook@latest automigrate --config-dir apps/wv1/.storybook --renderer @storybook/web-components`,
            `${pm.exec} storybook@latest automigrate --config-dir apps/ww1/.storybook --renderer @storybook/web-components`,
            `${pm.exec} storybook@latest automigrate --config-dir apps/ngapp/.storybook --renderer @storybook/angular`,
            `${pm.exec} storybook@latest automigrate --config-dir apps/ngapp-ts/.storybook --renderer @storybook/angular`,
            '',
            '4. Call the Nx generator to finish the migration:',
            'nx g @nx/storybook:migrate-7 --afterMigration',
          ],
          title: 'Storybook 7 Migration Guide',
        })
      );
    });
  });

  describe('removeViteTsConfigPathsPlugin', () => {
    it(`should remove ViteTsConfigPathsPlugin from main.js and the whole plugin array if empty`, () => {
      removeViteTsConfigPathsPlugin(tree, `apps/rv1/.storybook/main.js`);

      expect(
        tree.read('apps/rv1/.storybook/main.js', 'utf-8')
      ).toMatchSnapshot();
    });

    it(`should remove ViteTsConfigPathsPlugin from main.ts and the whole plugin array if empty`, () => {
      removeViteTsConfigPathsPlugin(tree, `apps/rv1/.storybook/main.js`);
      removeViteTsConfigPathsPlugin(tree, `apps/rv2-ts/.storybook/main.ts`);

      expect(
        tree.read('apps/rv2-ts/.storybook/main.ts', 'utf-8')
      ).toMatchSnapshot();
    });
  });

  describe('addViteConfigFilePathInFrameworkOptions', () => {
    it(`should add viteConfigFilePath in frameworkOptions in main.js`, () => {
      writeNewMainJs(tree);
      tree.write('apps/rv1/vite.config.js', '');
      addViteConfigFilePathInFrameworkOptions(
        tree,
        'apps/rv1/.storybook/main.js',
        'apps/rv1/vite.config.js'
      );

      expect(
        tree.read('apps/rv1/.storybook/main.js', 'utf-8')
      ).toMatchSnapshot();
    });

    it(`should add viteConfigFilePath in frameworkOptions in main.js if no options present`, () => {
      writeNewMainJs(tree);
      tree.write('apps/wv1/vite.config.js', '');
      addViteConfigFilePathInFrameworkOptions(
        tree,
        'apps/wv1/.storybook/main.js',
        'apps/wv1/vite.config.js'
      );

      expect(
        tree.read('apps/wv1/.storybook/main.js', 'utf-8')
      ).toMatchSnapshot();
    });
  });

  describe('changeCoreCommonImportToFramework', () => {
    it(`should change imports`, async () => {
      writeNewMainTs(tree);
      changeCoreCommonImportToFramework(tree, `apps/rv2-ts/.storybook/main.ts`);
      changeCoreCommonImportToFramework(
        tree,
        `apps/ngapp-ts/.storybook/main.ts`
      );

      expect(
        tree.read('apps/rv2-ts/.storybook/main.ts', 'utf-8')
      ).toMatchSnapshot();
      expect(
        tree.read('apps/ngapp-ts/.storybook/main.ts', 'utf-8')
      ).toMatchSnapshot();
    });
  });

  describe('removeUiFrameworkFromProjectJson', () => {
    it('should remove uiFramework from project.json', () => {
      expect(
        readProjectConfiguration(tree, 'rv1').targets['storybook']['options'][
          'uiFramework'
        ]
      ).toBeDefined();
      expect(
        readProjectConfiguration(tree, 'rv2-ts').targets['storybook'][
          'options'
        ]['uiFramework']
      ).toBeDefined();

      removeUiFrameworkFromProjectJson(tree);

      expect(
        readProjectConfiguration(tree, 'rv1').targets['storybook']['options'][
          'uiFramework'
        ]
      ).toBeUndefined();

      expect(
        readProjectConfiguration(tree, 'rv2-ts').targets['storybook'][
          'options'
        ]['uiFramework']
      ).toBeUndefined();
    });
  });

  describe('logResult', () => {
    it('should create the summary file with the correct content', () => {
      logResult(tree, {
        successfulProjects: {
          nextapp:
            'npx storybook@latest automigrate --config-dir apps/nextapp/.storybook --renderer @storybook/react',
          rv1: 'npx storybook@latest automigrate --config-dir apps/rv1/.storybook --renderer @storybook/react',
        },
        failedProjects: {
          'rv2-ts': `npx storybook@latest automigrate --config-dir apps/rv2-ts/.storybook --renderer @storybook/react`,
          rw1: 'npx storybook@latest automigrate --config-dir apps/rw1/.storybook --renderer @storybook/react',
        },
      });
      expect(
        tree.read('storybook-migration-summary.md', 'utf-8')
      ).toMatchSnapshot();
    });
  });
});

export function addAllProjectsToWorkspace(tree: Tree) {
  for (const [name, project] of Object.entries(allProjects)) {
    addProjectConfiguration(tree, name, project as ProjectConfiguration);
  }
  writeMainJs(tree);
  writeViteConfig(tree);
}

function writeViteConfig(tree: Tree) {
  tree.write(
    `apps/rv1/vite.config.js`,
    `const { defineConfig } = require('vite');`
  );

  tree.write(
    `apps/rv2-ts/vite.config.ts`,
    `const { defineConfig } = require('vite');`
  );

  tree.write(
    `apps/wv1/vite.config.ts`,
    `const { defineConfig } = require('vite');`
  );
}

function writeMainJs(tree: Tree) {
  tree.write(
    `apps/rv1/.storybook/main.js`,
    `
    const { mergeConfig } = require('vite');
    const viteTsConfigPaths = require('vite-tsconfig-paths').default;

    module.exports = {
      core: { builder: '@storybook/builder-vite' },
      stories: [
        '../src/app/**/*.stories.mdx',
        '../src/app/**/*.stories.@(js|jsx|ts|tsx)',
      ],
      addons: ['@storybook/addon-essentials'],
      async viteFinal(config, { configType }) {
        return mergeConfig(config, {
          plugins: [
            viteTsConfigPaths({
              root: '../../../',
            }),
          ],
        });
      },
    };
`
  );

  tree.write(
    `apps/rv2-ts/.storybook/main.ts`,
    `
    import type { StorybookConfig } from '@storybook/core-common';

    import { mergeConfig } from 'vite';
    import viteTsConfigPaths from 'vite-tsconfig-paths';

    const config: StorybookConfig = {
      core: { builder: '@storybook/builder-vite' },
      stories: [
        '../src/app/**/*.stories.mdx',
        '../src/app/**/*.stories.@(js|jsx|ts|tsx)',
      ],
      addons: ['@storybook/addon-essentials'],
      async viteFinal(config: any) {
        return mergeConfig(config, {
          plugins: [
            viteTsConfigPaths({
              root: '../../../',
            }),
          ],
        });
      },
    } as StorybookConfig;

    module.exports = config;
  `
  );

  tree.write(
    `apps/nextapp/.storybook/main.js`,
    `
    const path = require('path');

    module.exports = {
      core: { builder: 'webpack5' },
      stories: [
        '../components/**/*.stories.mdx',
        '../components/**/*.stories.@(js|jsx|ts|tsx)',
      ],
      addons: [
        '@storybook/addon-essentials',
        '@nx/react/plugins/storybook',
        'storybook-addon-swc',
        {
          name: 'storybook-addon-next',
          options: {
            nextConfigPath: path.resolve(__dirname, '../next.config.js'),
          },
        },
      ],
    };
  `
  );

  tree.write(
    `apps/nextapp-ts/.storybook/main.ts`,
    `
    import type { StorybookConfig } from '@storybook/core-common';
    import path from 'path';

    const config: StorybookConfig = {
      core: { builder: 'webpack5' },
      stories: [
        '../components/**/*.stories.mdx',
        '../components/**/*.stories.@(js|jsx|ts|tsx)',
      ],
      addons: [
        '@storybook/addon-essentials',
        '@nx/react/plugins/storybook',
        'storybook-addon-swc',
        {
          name: 'storybook-addon-next',
          options: {
            nextConfigPath: path.resolve(__dirname, '../next.config.js'),
          },
        },
      ],
    } as StorybookConfig;

    module.exports = config;
  `
  );
}

function writeNewMainJs(tree: Tree) {
  tree.write(
    `apps/rv1/.storybook/main.js`,
    `
    const { mergeConfig } = require('vite');

    module.exports = {
      stories: [
        '../src/app/**/*.stories.mdx',
        '../src/app/**/*.stories.@(js|jsx|ts|tsx)',
      ],
      addons: ['@storybook/addon-essentials'],
      async viteFinal(config, { configType }) {
        return mergeConfig(config, {});
      },
      framework: {
        name: '@storybook/react-vite',
        options: {},
      },
    };
    `
  );

  tree.write(
    `apps/wv1/.storybook/main.js`,
    `
    const { mergeConfig } = require('vite');

    module.exports = {
      stories: [
        '../src/app/**/*.stories.mdx',
        '../src/app/**/*.stories.@(js|jsx|ts|tsx)',
      ],
      addons: ['@storybook/addon-essentials'],
      async viteFinal(config, { configType }) {
        return mergeConfig(config, {});
      },
      framework: {
        name: '@storybook/web-components-vite',
      },
    };
    `
  );
}

function writeNewMainTs(tree: Tree) {
  tree.write(
    `apps/rv2-ts/.storybook/main.ts`,
    `
    import type { StorybookConfig } from '@storybook/core-common';
    import { mergeConfig } from 'vite';
    const config: StorybookConfig = {
      core: {},
      stories: ['../src/app/**/*.stories.mdx', '../src/app/**/*.stories.@(js|jsx|ts|tsx)'],
      addons: ['@storybook/addon-essentials'],
      async viteFinal(config: any) {
        return mergeConfig(config, {});
      },
      framework: {
        name: '@storybook/react-vite',
        options: {}
      }
    };
    module.exports = config;
    `
  );

  tree.write(
    `apps/ngapp-ts/.storybook/main.ts`,
    `
    import type { StorybookConfig } from '@storybook/core-common';
    const config: StorybookConfig = {
      core: {},
      stories: ['../src/app/**/*.stories.mdx', '../src/app/**/*.stories.@(js|jsx|ts|tsx)'],
      addons: ['@storybook/addon-essentials'],
      framework: {
        name: '@storybook/angular',
        options: {}
      }
    };
    module.exports = config;
    `
  );
}
