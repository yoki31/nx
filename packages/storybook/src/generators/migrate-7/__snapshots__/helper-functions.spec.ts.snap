// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helper functions for the Storybook 7 migration generator addViteConfigFilePathInFrameworkOptions should add viteConfigFilePath in frameworkOptions in main.js 1`] = `
"
    const { mergeConfig } = require('vite');

    module.exports = {
      stories: [
        '../src/app/**/*.stories.mdx',
        '../src/app/**/*.stories.@(js|jsx|ts|tsx)',
      ],
      addons: ['@storybook/addon-essentials'],
      async viteFinal(config, { configType }) {
        return mergeConfig(config, {});
      },
      framework: {
        name: '@storybook/react-vite',
        options: {
                      builder: {
                        viteConfigPath: 'apps/rv1/vite.config.js',
                      },
                      },
      },
    };
    "
`;

exports[`Helper functions for the Storybook 7 migration generator addViteConfigFilePathInFrameworkOptions should add viteConfigFilePath in frameworkOptions in main.js if no options present 1`] = `
"
    const { mergeConfig } = require('vite');

    module.exports = {
      stories: [
        '../src/app/**/*.stories.mdx',
        '../src/app/**/*.stories.@(js|jsx|ts|tsx)',
      ],
      addons: ['@storybook/addon-essentials'],
      async viteFinal(config, { configType }) {
        return mergeConfig(config, {});
      },
      framework: { options: {
                      builder: {
                        viteConfigPath: 'apps/wv1/vite.config.js',
                      },
                    },
                      
        name: '@storybook/web-components-vite',
      },
    };
    "
`;

exports[`Helper functions for the Storybook 7 migration generator changeCoreCommonImportToFramework should change imports 1`] = `
"
    import type { StorybookConfig } from '@storybook/react-vite';
    import { mergeConfig } from 'vite';
    const config: StorybookConfig = {
      core: {},
      stories: ['../src/app/**/*.stories.mdx', '../src/app/**/*.stories.@(js|jsx|ts|tsx)'],
      addons: ['@storybook/addon-essentials'],
      async viteFinal(config: any) {
        return mergeConfig(config, {});
      },
      framework: {
        name: '@storybook/react-vite',
        options: {}
      }
    };
    module.exports = config;
    "
`;

exports[`Helper functions for the Storybook 7 migration generator changeCoreCommonImportToFramework should change imports 2`] = `
"
    import type { StorybookConfig } from '@storybook/angular';
    const config: StorybookConfig = {
      core: {},
      stories: ['../src/app/**/*.stories.mdx', '../src/app/**/*.stories.@(js|jsx|ts|tsx)'],
      addons: ['@storybook/addon-essentials'],
      framework: {
        name: '@storybook/angular',
        options: {}
      }
    };
    module.exports = config;
    "
`;

exports[`Helper functions for the Storybook 7 migration generator getAllStorybookInfo and onlyShowGuide should return all info for all projects with Storybook 1`] = `
{
  "nextapp": {
    "configDir": "apps/nextapp/.storybook",
    "uiFramework": "@storybook/react",
    "viteConfigFilePath": undefined,
  },
  "nextapp-ts": {
    "configDir": "apps/nextapp-ts/.storybook",
    "uiFramework": "@storybook/react",
    "viteConfigFilePath": undefined,
  },
  "ngapp": {
    "configDir": "apps/ngapp/.storybook",
    "uiFramework": "@storybook/angular",
  },
  "ngapp-ts": {
    "configDir": "apps/ngapp-ts/.storybook",
    "uiFramework": "@storybook/angular",
  },
  "rv1": {
    "configDir": "apps/rv1/.storybook",
    "uiFramework": "@storybook/react",
    "viteConfigFilePath": "apps/rv1/vite.config.js",
  },
  "rv2-ts": {
    "configDir": "apps/rv2-ts/.storybook",
    "uiFramework": "@storybook/react",
    "viteConfigFilePath": "apps/rv2-ts/vite.config.ts",
  },
  "rw1": {
    "configDir": "apps/rw1/.storybook",
    "uiFramework": "@storybook/react",
    "viteConfigFilePath": undefined,
  },
  "wv1": {
    "configDir": "apps/wv1/.storybook",
    "uiFramework": "@storybook/web-components",
    "viteConfigFilePath": "apps/wv1/vite.config.ts",
  },
  "ww1": {
    "configDir": "apps/ww1/.storybook",
    "uiFramework": "@storybook/web-components",
    "viteConfigFilePath": undefined,
  },
}
`;

exports[`Helper functions for the Storybook 7 migration generator logResult should create the summary file with the correct content 1`] = `
"# Storybook 7 Migration Summary

## Upgrade Storybook packages

The following command was ran to upgrade the Storybook packages:

\`\`\`bash
npx storybook@latest upgrade
\`\`\`

## Your \`.storybook/main.js|ts\` files were prepared for Storybook's automigration scripts

Some adjustments were made to your \`.storybook/main.js|ts\` files so that
the Storybook automigration scripts could run successfully. The changes that were made are as follows:

- Remove the \`as StorybookConfig\` typecast from the main.ts files, if any,
  since it is not needed any more.
- Remove the \`path.resolve\` calls from the Next.js Storybook configuration, if any, since it breaks the Storybook automigration scripts.

## The Storybook automigration scripts were ran


The following commands ran successfully and your Storybook configuration was successfully migrated to the latest version 7:


- \`npx storybook@latest automigrate --config-dir apps/nextapp/.storybook --renderer @storybook/react\`

- \`npx storybook@latest automigrate --config-dir apps/rv1/.storybook --renderer @storybook/react\`


Please make sure to check the results yourself and make sure that everything is working as expected.



### Some migrations failed

The following commands failed and your Storybook configuration for these projects was not
migrated to the latest version 7:


- \`npx storybook@latest automigrate --config-dir apps/rv2-ts/.storybook --renderer @storybook/react\`

- \`npx storybook@latest automigrate --config-dir apps/rw1/.storybook --renderer @storybook/react\`


You can run these commands again, manually, and follow the instructions in the
output of these commands to migrate your Storybook configuration to the latest version 7.


Also, we may have missed something. Please make sure to check the logs of the Storybook CLI commands that were run, and look for
the \`❌ Failed trying to evaluate\` message or \`❌ The migration failed to update\` message. This will indicate if a command was
unsuccessful, and will help you run the migration again, manually.

## Final adjustments

After the Storybook automigration scripts have run, some additional adjustments were made to your
workspace, to make sure that everything is working as expected. These adjustments are as follows:

- The \`vite-tsconfig-paths\` plugin was removed from the Storybook configuration files since it's no longer needed.
- The \`viteConfigPath\` option was added to the Storybook builder, where needed.
- The import package for the \`StorybookConfig\` type was changed to be framework specific.
- The \`uiFramework\` option was removed from your project's Storybook targets.
- The \`lit\` package was added to your workspace, if you are using the
Web Components \`@storybook/web-components\` package. Please note that the \`lit-html\` package is
no longer needed by Storybook v7. So, if you are not using it anywhere else, you can safely remove it.

## Next steps

You can make sure everything is working as expected by trying
to build or serve your Storybook as you normally would.

\`\`\`bash
npx nx build-storybook project-name
\`\`\`

\`\`\`bash
npx nx storybook project-name
\`\`\`

Please read the [Storybook 7.0.0 release article](https://storybook.js.org/blog/storybook-7-0/) and the
official [Storybook 7.0.0 migration guide](https://storybook.js.org/docs/react/migration-guide)
for more information.

You can also read the docs for the [@nx/storybook:migrate-7 generator](https://nx.dev/nx-api/storybook/generators/migrate-7) and our [Storybook 7 setup guide](https://nx.dev/nx-api/storybook/documents/storybook-7-setup).
"
`;

exports[`Helper functions for the Storybook 7 migration generator removePathResolvesFromNextConfig should remove path from nextConfigPath in main.js 1`] = `
"
    const path = require('path');

    module.exports = {
      core: { builder: 'webpack5' },
      stories: [
        '../components/**/*.stories.mdx',
        '../components/**/*.stories.@(js|jsx|ts|tsx)',
      ],
      addons: [
        '@storybook/addon-essentials',
        '@nx/react/plugins/storybook',
        'storybook-addon-swc',
        {
          name: 'storybook-addon-next',
          options: {
            nextConfigPath: '../next.config.js',
          },
        },
      ],
    };
  "
`;

exports[`Helper functions for the Storybook 7 migration generator removePathResolvesFromNextConfig should remove path from nextConfigPath in main.js 2`] = `
"
    import type { StorybookConfig } from '@storybook/core-common';
    import path from 'path';

    const config: StorybookConfig = {
      core: { builder: 'webpack5' },
      stories: [
        '../components/**/*.stories.mdx',
        '../components/**/*.stories.@(js|jsx|ts|tsx)',
      ],
      addons: [
        '@storybook/addon-essentials',
        '@nx/react/plugins/storybook',
        'storybook-addon-swc',
        {
          name: 'storybook-addon-next',
          options: {
            nextConfigPath: '../next.config.js',
          },
        },
      ],
    } as StorybookConfig;

    module.exports = config;
  "
`;

exports[`Helper functions for the Storybook 7 migration generator removeTypecastFromMainTs should remove typecast from TypeScript files 1`] = `
"
    import type { StorybookConfig } from '@storybook/core-common';

    import { mergeConfig } from 'vite';
    import viteTsConfigPaths from 'vite-tsconfig-paths';

    const config: StorybookConfig = {
      core: { builder: '@storybook/builder-vite' },
      stories: [
        '../src/app/**/*.stories.mdx',
        '../src/app/**/*.stories.@(js|jsx|ts|tsx)',
      ],
      addons: ['@storybook/addon-essentials'],
      async viteFinal(config: any) {
        return mergeConfig(config, {
          plugins: [
            viteTsConfigPaths({
              root: '../../../',
            }),
          ],
        });
      },
    } ;

    module.exports = config;
  "
`;

exports[`Helper functions for the Storybook 7 migration generator removeTypecastFromMainTs should remove typecast from TypeScript files 2`] = `
"
    import type { StorybookConfig } from '@storybook/core-common';
    import path from 'path';

    const config: StorybookConfig = {
      core: { builder: 'webpack5' },
      stories: [
        '../components/**/*.stories.mdx',
        '../components/**/*.stories.@(js|jsx|ts|tsx)',
      ],
      addons: [
        '@storybook/addon-essentials',
        '@nx/react/plugins/storybook',
        'storybook-addon-swc',
        {
          name: 'storybook-addon-next',
          options: {
            nextConfigPath: path.resolve(__dirname, '../next.config.js'),
          },
        },
      ],
    } ;

    module.exports = config;
  "
`;

exports[`Helper functions for the Storybook 7 migration generator removeViteTsConfigPathsPlugin should remove ViteTsConfigPathsPlugin from main.js and the whole plugin array if empty 1`] = `
"
    const { mergeConfig } = require('vite');
    

    module.exports = {
      core: { builder: '@storybook/builder-vite' },
      stories: [
        '../src/app/**/*.stories.mdx',
        '../src/app/**/*.stories.@(js|jsx|ts|tsx)',
      ],
      addons: ['@storybook/addon-essentials'],
      async viteFinal(config, { configType }) {
        return mergeConfig(config, {
          
        });
      },
    };
"
`;

exports[`Helper functions for the Storybook 7 migration generator removeViteTsConfigPathsPlugin should remove ViteTsConfigPathsPlugin from main.ts and the whole plugin array if empty 1`] = `
"
    import type { StorybookConfig } from '@storybook/core-common';

    import { mergeConfig } from 'vite';
    

    const config: StorybookConfig = {
      core: { builder: '@storybook/builder-vite' },
      stories: [
        '../src/app/**/*.stories.mdx',
        '../src/app/**/*.stories.@(js|jsx|ts|tsx)',
      ],
      addons: ['@storybook/addon-essentials'],
      async viteFinal(config: any) {
        return mergeConfig(config, {
          
        });
      },
    } as StorybookConfig;

    module.exports = config;
  "
`;
